### PR TITLE
Feature: reload endpoint for hot load changes from config file

### PR DESCRIPTION
**Pull request by antoniomerlin**
In this PR an endpoint is added for loading hot changes without restarting the container for #346 . Just making POST request to `/-/reload` latest config changes are available in metrics.

Let me know if any changes are required in this.

### CODE REVIEW

File: cmd/yace/main.go [hunk 1]
```diff
@@ -170,3 +171,22 @@ func (s *scraper) scrape(ctx context.Context) (err error) {
 	log.Debug("Metrics scraped.")
 	return nil
 }
+
+func (s *scraper) reload(ctx context.Context) func(http.ResponseWriter, *http.Request){
```
Comment (thomaspeitz) [inline cmd/yace/main.go:175]: Lets put this function into the pkg/config.go. Would like to keep main.go small.

Can you please verify if the code was run in gofmt? https://pkg.go.dev/cmd/gofmt

**Review by thomaspeitz — CHANGES_REQUESTED** [review]
Thanks for your PR :) lets make some small changes and get this merged.

**Comment by thomaspeitz** [comment]
@antoniomerlin If you get this fixed till Thuersday morning, I get this into the next release.

Else I schedule it for next week :)

Greetings :)

File: cmd/yace/main.go (hunk 1 above)
Comment (antoniomerlin) [reply cmd/yace/main.go]: Hi thanks for reviewing, yes the code is run in gofmt

**Comment by antoniomerlin** [comment]
Hi i removed that reload as separate function and added directly inside http handlefunc, as for change requested to move this function inside config.go currently the way i am doing this is loading the config file again and calling the scrape function for making the config changes available in next metric scrape.
```
log.Println("Parse config..")
if err := config.Load(configFile); err != nil {
	log.Fatal("Couldn't read ", *configFile, ": ", err)
}
if *decoupledScraping {
	s.decoupled(ctx)
} else {
	s.scrape(ctx)
}
```
What are your thought about this approach

**Pull request merged**